tad_pathways/tad.py

```diff
@@ -0,0 +1,189 @@
+
+Usage: Import only
+
+    from tad import TADPathways
+"""
+
+import os
+import subprocess
+
+from tad_pathways.genes.genelist import TADGeneList
+
+
+class TadPathways:
+    """
+    Class methods to perform a tad_pathways analysis
+    """
+
+    def __init__(
+        self,
+        snp_list_name,
+        snp_list_location,
+        base_dir,
+        remove_hla=False,
+        all_pathways=False,
+    ):
+        """
+        Arguments:
+        snp_list_name - the name of the input snp list
+        base_dir - the root directory of where to save files
+        remove_hla - boolean if HLA TAD should be removed
+        """
+        self.snp_list_name = snp_list_name
+        self.snp_list_location = snp_list_location
+        self.base_dir = base_dir
+        self.remove_hla = remove_hla
+        self.all_pathways = all_pathways
+        self._create_filenames()
+
+    def _create_filenames(self):
+        """
+        Create several variables that correspond to the specific snp list
+        """
+
+        if not os.path.exists(self.base_dir):
+            os.makedirs(self.base_dir)
+
+        # File to compile RSid lookup results
+        self.snp_location_file = os.path.join(
+            self.base_dir, "{}_location.tsv".format(self.snp_list_name)
+        )
+
+        # Files for TAD based genesets - all genes that fall in candidate variant TADs
+        self.genelist_file = os.path.join(
+            self.base_dir, "{}_tad_genelist.tsv".format(self.snp_list_name)
+        )
+
+        # Files from WebGestalt output
+        self.pathway_p_values_file = os.path.join(
+            self.base_dir, "{}_pvals.tsv".format(self.snp_list_name)
+        )
+
+        # Files used to summarize results
+        self.nearest_gene_file = os.path.join(
+            self.base_dir, "{}_tad_genelist_nearest_gene.tsv".format(self.snp_list_name)
+        )
+
+        self.gene_evidence_file = os.path.join(
+            self.base_dir, "{}_gene_evidence.csv".format(self.snp_list_name)
+        )
+
+        self.evidence_summary_file = os.path.join(
+            self.base_dir, "{}_gene_evidence_summary.tsv".format(self.snp_list_name)
+        )
+
+        self.all_pathways_evidence = os.path.join(
+            self.base_dir,
+            "{}_all-sig-pathways_gene_evidence.csv".format(self.snp_list_name),
+        )
+
+        self.all_pathways_summary = os.path.join(
+            self.base_dir,
+            "{}_all-sig-pathways_gene_evidence_summary.tsv".format(self.snp_list_name),
+        )
+
+    def build_snp_list(self):
+        command_list = [
+            "Rscript",
+            "--vanilla",
+            "scripts/build_snp_list.R",
+            "--snp_file",
+            self.snp_list_location,
+            "--output_file",
+            self.snp_location_file,
+        ]
+        subprocess.call(command_list)
+
+    def build_custom_tad_genelist(self):
+        command_list = [
+            "python",
```

Comment:
Is `scripts/build_custom_tad_genelist.py` compatible for Python 2 or 3 (or both)? By default, `python` command
calls python 2 on many (if not most) operating systems.

Reply:
python 3 - should I change the call to `"python3"`?

Reply:
If this package is for Python 3 only, then it is a good idea to change it to `python3`,

Reply:
done in fbe303d09912f75b79f408e6a9bc0a0419bf134a